node/middlewares/newSitemap.ts

```diff
@@ -1,33 +1,38 @@
+import { VBase } from '@vtex/api'
 import * as cheerio from 'cheerio'
-import { keys, map, mergeAll, prop, range, replace } from 'ramda'
+import { replace } from 'ramda'
 
-import { Internal, RouteIndexFileEntry } from '../clients/rewriter'
-import { currentDate } from '../resources/utils'
+import { Internal } from '../clients/rewriter'
+import {
+  GENERATE_SITEMAP_EVENT,
+  SITEMAP_BUCKET,
+  SITEMAP_INDEX,
+  SitemapEntry,
+  SitemapIndex,
+} from './generateSitemap'
 
-const MAX_ROUTES_PER_REQUEST = 100
-const TEN_MINUTES_S = 10 * 60
+const ONE_DAY_S = 24 * 60 * 60
 
-const sitemapIndexEntry = (forwardedHost: string, rootPath: string, entity: string, firstIndex?: number, lastIndex?: number): string => {
-  if (firstIndex && lastIndex) {
-    return `
-      <sitemap>
-        <loc>https://${forwardedHost}${rootPath}/_v/public/newsitemap/${entity}-${firstIndex}-${lastIndex}.xml</loc>
-        <lastmod>${currentDate()}</lastmod>
-      </sitemap>
-    `
-  }
-  return `
-    <sitemap>
-      <loc>https://${forwardedHost}${rootPath}/_v/public/newsitemap/${entity}.xml</loc>
-      <lastmod>${currentDate()}</lastmod>
-    </sitemap>
-  `
-}
+const sitemapIndexEntry = (
+  forwardedHost: string,
+  rootPath: string,
+  entry: string,
+  lastUpdated: string
+) =>
+  `<sitemap>
+      <loc>https://${forwardedHost}${rootPath}/_v/public/newsitemap/${entry}.xml</loc>
+      <lastmod>${lastUpdated}</lastmod>
+    </sitemap>`
 
-const URLEntry = (forwardedHost: string, rootPath: string, route: Internal): string => {
+const URLEntry = (
+  forwardedHost: string,
+  rootPath: string,
+  route: Internal,
+  lastUpdated: string
+): string => {
   let entry = `
       <loc>https://${forwardedHost}${rootPath}${route.from}</loc>
-      <lastmod>${currentDate()}</lastmod>
+      <lastmod>${lastUpdated}</lastmod>
```

Comment:
I get it here, but I still believe this date would be better to be the date where this route/product has received any change.